### ISSUE TITLE
NaNs in LensUI 2.19 instead of timings

### ISSUE THREAD

**Issue by f0y** [openzipkin/zipkin#2906]
## Describe the Bug
There is a bug with trace rendering in LensUI 2.19. I see NaNs instead of timings and there are no spans displayed.
 
![image](https://user-images.githubusercontent.com/608714/68178377-daa8fe00-ff9c-11e9-84ae-94e9d37a2639.png)

## Steps to Reproduce
Upload trace https://gist.github.com/f0y/1fd8461bd52c522acd71061ebeff965e

**Comment by tacigar** [comment]
Thank you for your report!
If timestamps are not set for some spans, it seems that UI cannot handle it well.
I'll fix this bug as soon as possible.

**Comment by shakuzen** [comment]
> If timestamps are not set for some spans, it seems that UI cannot handle it well.

I agree it isn't good for the UI to fail to render entirely, but I'm wondering if there's an instrumentation bug or some other reason why some of the spans don't have a timestamp. From the [API documentation](https://zipkin.io/zipkin-api/#/definitions/Span) on the `timestamp` field:

> Epoch microseconds of the start of this span, possibly absent if
> incomplete.
> 
> For example, 1502787600000000 corresponds to 2017-08-15 09:00 UTC
> 
> This value should be set directly by instrumentation, using the most
> precise value possible. For example, gettimeofday or multiplying epoch
> millis by 1000.
> 
> There are three known edge-cases where this could be reported absent.
> 
>   * A span was allocated but never started (ex not yet received a timestamp)
>   * The span’s start event was lost
>   * Data about a completed span (ex tags) were sent after the fact

Maybe this is the type of situation we should give an indication in the UI that the data should be looked at to determine why it is missing generally expected elements.

**Comment by jcchavezs** [comment]
Agree with Tommy. This is not normal situation, this could happen either in a bug in instrumentation or a misuse of the tracer (I have seen some instrumentation creating spans for things that should be logs). In either case we should warn the user.

**Comment by codefromthecrypt** [comment]
there are other data problems besides missing timestamps. I think before we just avoided things like divide by zero problems by skipping calculation of those spans. if no span has a timestamp we could error out. if only some are missing a timestamp we can still present the rest of the trace I think.

**Comment by codefromthecrypt** [comment]
iotw, my opinion is to just avoid doing bad math. 

imho whether to warn or not is separate and optional, and if we decide to only warn about timestamps is that because we are only thinking about that? or we would only warn about timestamps and not things like backwards endpoints, flipped client/server or the various other problems? maybe just start with not bombing is better than having a one-off solution to one of many problems data can have.
